cinema/views.py

```diff
@@ -56,3 +78,46 @@ def get_serializer_class(self):
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        date = self.request.query_params.get("date")
+        movie = self.request.query_params.get("movie")
+
+        if date:
+            queryset = queryset.filter(show_time__date=date)
+
+        if movie:
+            queryset = queryset.filter(movie=movie)
+
+        if self.action == "list":
+            queryset = (
+                queryset
+                .select_related("movie", "cinema_hall")
+                .annotate(
+                    tickets_available=(
+                        F("cinema_hall__rows") * F("cinema_hall__seats_in_row")
+                    ) - Count("tickets")
```

Comment:
Ensure that the `tickets_available` calculation aligns with the logic in the `MovieSessionListSerializer`. This annotation subtracts the count of tickets from the total capacity, which should match the serializer's expectations.